components/ai_chat/content/browser/ai_chat_tab_helper.cc

```diff
@@ -185,6 +218,49 @@ void AIChatTabHelper::OnFaviconUpdated(
   OnFaviconImageDataChanged();
 }
 
+// mojom::PageContentExtractorHost
+void AIChatTabHelper::OnInterceptedPageContentChanged(
+    mojom::PageContentPtr content) {
+  DVLOG(2) << __func__;
+
+  // TODO(petemill): Only accept basic content string - no fetching or even json
+  // parsing. Store in local variable until getpagecontent is called. This is
+  // performance optimization so that we're not doing extra work when there's no
+  // active conversation. We still need to store this data for inactive
+  // conversations so that if a conversation is opened for the current
+  // navigation then we are able to retrieve the content that's only accessible
+  // via subresource interception.
```

Comment:
Do not merge until this is resolved. Otherwise we'll be fetching the transcript content at the provided Urls on every relevant navigation, even when there's no active AI Chat conversation

Reply:
This is resolved now - nothing will be parsed (or network fetched) until an AI Chat message is sent